### DISCUSSION TITLE
Bulk Operations: Locales & Query

### DISCUSSION THREAD

**Discussion by unonweb** [payloadcms/payload#2767] [Q&A]
Hi,
bulk operations work great, good job!
I just realized that when requested by a bulk operation `req.locale` doesn't contain the current locale anymore.
I could mitigate that issue if I could know by any property in `req` that the query comes from a bulk operation. Is there a way to detect that?

**Comment by DanRibbens** [comment]
Hi @unonweb,
I looked into solving the underlying missing `req.locale` and I don't see where it could be missing. The same middleware that sets the locale is run for all REST requests and the local operations all seem to assign req.locale the same way.

Where is it that you're not seeing req.locale?

I would love to see this reproduced so we can fix it!

**Comment by unonweb** [comment]
`req.locale` is there but it doesn't contain the currently selected locale. As you can see in the following screenshots `req.locale` doesn't change its value when I change the language environment in the admin panel. Doing this in non-bulk operations behaves like expected.
### locale = de (admin panel)
![lang=de (admin)](https://github.com/payloadcms/payload/assets/79059627/848efbff-ac2e-4c5e-bc4a-9555d806f595)
### locale = de (code)
![lang=de (code)](https://github.com/payloadcms/payload/assets/79059627/dba2891b-e443-46c3-a798-801caa007ce9)
### locale = en (admin panel)
![lang=en (admin)](https://github.com/payloadcms/payload/assets/79059627/f524a17d-50a2-4fa5-b730-857214ca7b8e)
### locale = en (code)
![lang=en (code)](https://github.com/payloadcms/payload/assets/79059627/cc54e645-0815-47f1-bb18-de2086d6dcbc)

**Reply by DanRibbens** [reply]
Now I understand. I was able to reproduce. I am working on a fix now.

**Comment by DanRibbens** [comment]
I've created the issue https://github.com/payloadcms/payload/issues/2778 from this discussion and went ahead and opened a PR that fixes it:

https://github.com/payloadcms/payload/pull/2779

Thank you @unonweb for reporting and providing extra detail!

We'll get this merged and released this week.
